### DISCUSSION TITLE
What is the significance of falling and rising edges in the OpenROAD GUI?

### DISCUSSION THREAD

**Discussion by oharboe** [The-OpenROAD-Project/OpenROAD#3673] [General]
For `designs/asap7/gcd/config.mk` (default), I ran `make synth` and then `make gui_synth`.

What is the significance and origin of the falling and rising edges in the GUI?

![image](https://github.com/The-OpenROAD-Project/OpenROAD/assets/2798822/977abf34-29d6-4e0e-b5cf-49ce396ed1e2)

**Comment by maliberty** [comment]
It tells you whether the signal is rising or falling in the path being shown.

**Reply by oharboe** [reply]
What is the origin of it going back and forth between rising and falling edge above?

**Reply by maliberty** [reply]
Most cmos gates are inverting (inv, nand, nor)

**Reply by oharboe** [reply]
Still confused: the value of the signal can only be known dynamically. So in what sense is it falling or rising from a static timing point of view?

**Reply by maliberty** [reply]
sta is considering all cases of rising/falling that are statically possible and reporting on a case that fails.

**Reply by rovinski** [reply]
The transition simply shows that for this timing path, the timing in question is caused by the signal rising or falling at this pin. Cells almost always have different rise and fall times so it is sometimes useful to show which one is occurring.

In this timing path, for example, The output pin of the FF (`_942_/Q`) is rising, which causes a rising edge at the inverter (`_529_/A`), which causes a falling edge at the nand (`_530_/A1`).

If you have an inverting gate, (inv, nand, nor, etc) the output transition is going to be opposite of the input transition. If you have a non-inverting gate (buf, and, or, etc.) then the output transition will match the input transition. If you see many signals oscillating through the timing path, it's likely for the simple reason that inverting gates are more frequently due to being faster and smaller.

**Reply by rovinski** [reply]
> the value of the signal can only be known dynamically

STA looks at the worst cases. It can look at other cases too but we generally only care about the worst case. STA has calculated that this sequence of transitions causes the worst possible timing along this path.
